src/main/java/uk/gov/justice/laa/crime/cfecrime/utils/RequestHandler.java

```diff
@@ -2,44 +2,98 @@
+        if (cfeCrimeRequest.getSectionPassportedBenefit() != null) {
+            passported = cfeCrimeRequest.getSectionPassportedBenefit().getPassportedBenefit();
         }
-        oc = getOutcomeFromAgeAndPassportedBenefit(under18,passported);
+        outcome = getOutcomeFromAgeAndPassportedBenefit(under18, passported);
+
+        CfeCrimeResponse cfeCrimeResponse = new CfeCrimeResponse();
+        if (outcome != null) {
+            cfeCrimeResponse.setOutcome(outcome);
+        }else{
+            //call cma
+            statelessApiResponse = cmaService.callCma(cmaRequest);
+
+            setInitialMeansTestOutcome(cfeCrimeRequest, cfeCrimeResponse);
+            if (statelessApiResponse.getInitialMeansAssessment().getResult() == InitAssessmentResult.FULL) {
+                setFullMeansTestOutcome(cfeCrimeRequest, cfeCrimeResponse);
+            }
 
-        CfeCrimeResponse response = new CfeCrimeResponse();
-        if (oc != null) {
-            Result result = new Result();
-            result.setOutcome(oc);
-            response.withResult(result);
         }
-        return response;
+        return cfeCrimeResponse;
     }
 
-    private static Outcome getOutcomeFromAgeAndPassportedBenefit(Boolean clientUnder18, Boolean clientPassportedBenefit){
+    private static Outcome getOutcomeFromAgeAndPassportedBenefit(Boolean clientUnder18, Boolean clientPassportedBenefit) {
         Outcome outcome = null;
-        if (clientUnder18 != null && clientUnder18.booleanValue()){
-            outcome = Result.Outcome.ELIGIBLE;
+        if (clientUnder18 != null && clientUnder18.booleanValue()) {
+            outcome = Outcome.ELIGIBLE;
         }
-        if (clientPassportedBenefit !=null && clientPassportedBenefit.booleanValue()){
-            outcome = Result.Outcome.ELIGIBLE;
+        if (clientPassportedBenefit != null && clientPassportedBenefit.booleanValue()) {
+            outcome = Outcome.ELIGIBLE;
         }
         return outcome;
     }
+
+    public static void setFullMeansTestOutcome(CfeCrimeRequest cfeCrimeRequest, CfeCrimeResponse cfeCrimeResponse) {
+
+        Outcome fullOutcome = null;
+
+        if (cfeCrimeRequest.getSectionFullMeansTest() != null){
+            FullAssessmentResult statelessInitialResult = statelessApiResponse.getFullMeansAssessment().getResult();
+            fullOutcome = FullMeansTestOutcomeCalculator.getFullMeansTestOutcome(statelessInitialResult, caseType, magCourtOutcome);
+            cfeCrimeResponse.setOutcome(fullOutcome);
+        }
+        cfeCrimeResponse.setOutcome(fullOutcome);
+
+    }
+
+    public static void setInitialMeansTestOutcome(CfeCrimeRequest cfeCrimeRequest, CfeCrimeResponse cfeCrimeResponse) throws UndefinedOutcomeException {
+
+        Outcome initOutcome = null;
+        if (cfeCrimeRequest.getSectionInitialMeansTest() != null) {
+            caseType = CaseType.valueOf(cfeCrimeRequest.getSectionInitialMeansTest().getCaseType().name());
```

Comment:
surely this is just cfeCrimeRequest.getCaseType() ? [There shouldn't be a case type in the initial means test section, its per-assessment]

Reply:
It needs a String not object, gives a compilation error expecting string.